pkg/services/rpcsrv/server_test.go

```diff
@@ -2003,29 +2213,82 @@ func testRPCProtocol(t *testing.T, doRPCCall func(string, string, *testing.T) []
 			tx.NetworkFee += netFee
 			size += sizeDelta
 			tx.NetworkFee += int64(size) * chain.FeePerByte()
+			tx.NetworkFee *= 2
 		}
 
-		newTx := func() *transaction.Transaction {
+		newTx := func(code opcode.Opcode, validUntilIncr uint32, systemFee int64, addNetFee bool, addAttrNotValidBeforeT bool, addAttrHighPriority bool) *transaction.Transaction {
 			height := chain.BlockHeight()
-			tx := transaction.New([]byte{byte(opcode.PUSH1)}, 0)
+			tx := transaction.New([]byte{byte(code)}, 0)
 			tx.Nonce = height + 1
-			tx.ValidUntilBlock = height + 10
+			tx.ValidUntilBlock = height + validUntilIncr
 			tx.Signers = []transaction.Signer{{Account: acc0.PrivateKey().GetScriptHash()}}
-			addNetworkFee(tx)
+			if addNetFee {
+				addNetworkFee(tx)
+			}
+			tx.SystemFee = systemFee
+			if addAttrNotValidBeforeT || addAttrHighPriority {
+				tx.Attributes = []transaction.Attribute{}
+				if addAttrNotValidBeforeT {
+					tx.Attributes = append(tx.Attributes, transaction.Attribute{Type: transaction.NotValidBeforeT, Value: &transaction.NotValidBefore{Height: height + 1}})
+				}
+				if addAttrHighPriority {
+					tx.Attributes = append(tx.Attributes, transaction.Attribute{Type: transaction.HighPriority})
+				}
+			}
 			require.NoError(t, acc0.SignTx(testchain.Network(), tx))
 			return tx
 		}
 
+		t.Run("non-transaction parameter", func(t *testing.T) {
+			body := doRPCCall(`{"jsonrpc": "2.0", "id": 1, "method": "calculatenetworkfee", "params": ["bm90IGEgdHJhbnNhY3Rpb24K"]}"`, httpSrv.URL, t)
+			_ = checkErrGetResult(t, body, true, neorpc.InvalidParamsCode, "Invalid Params")
+		})
```

Comment:
This is a separate test not related to the "submit" group.

Reply:
Fixed